src/Credentials/CredentialProvider.php

```diff
@@ -267,6 +265,27 @@ public static function ecsCredentials(array $config = [])
     }
 
     /**
+     * Remote credentials provider returns a credentials provider for the
+     * default remote endpoints such as EC2 or ECS Roles.
+     *
+     * @param array $config Array of configuration data.
+     *
+     * @return InstanceProfileProvider|EcsCredentialProvider
+     * @see Aws\Credentials\InstanceProfileProvider for $config details.
+     * @see Aws\Credentials\EcsCredentialProvider for $config details.
+     */
+    public static function remoteCredentialsProvider(array $config = [])
+    {
+        $ecsCredentialsUri = getenv(EcsCredentialProvider::ENV_URI);
+
+        if (!$ecsCredentialsUri) {
+            return self::instanceProfile($config);
+        }
+
+        return self::ecsCredentials($config);
```

Comment:
There is a problem with the logic here, even with `ENV` existing, there is possibility that the ecs credential fails when making the request, in that case, instanceProfile needs to be retrieved.